src/server/snapshot.cc

```diff
@@ -44,27 +46,32 @@ void SliceSnapshot::Start(bool include_journal_changes) {
   snapshot_version_ = db_slice_->RegisterOnChange(move(on_change));
   VLOG(1) << "DbSaver::Start - saving entries with version less than " << snapshot_version_;
 
-  if (include_journal_changes) {
+  if (include_journal) {
     auto* journal = db_slice_->shard_owner()->journal();
     DCHECK(journal);
-    journal_cb_id_ = journal->RegisterOnChange(
-        [this](const journal::Entry& e) { OnJournalEntry(e); });
+    journal_cb_id_ =
+        journal->RegisterOnChange([this](const journal::Entry& e) { OnJournalEntry(e); });
   }
 
   sfile_.reset(new io::StringFile);
 
   rdb_serializer_.reset(new RdbSerializer(sfile_.get()));
 
-  fb_ = fiber([this] {
-    FiberFunc();
-    db_slice_->UnregisterOnChange(snapshot_version_);
-    if (journal_cb_id_)
-      db_slice_->shard_owner()->journal()->Unregister(journal_cb_id_);
+  fb_ = fiber([this, auto_close] {
+    SerializeEntries();
+    if (auto_close)
+      Close();
   });
 }
 
-void SliceSnapshot::Join() {
+void SliceSnapshot::Stop() {
   fb_.join();
+  Close();
+}
```

Comment:
In theory, we can make it not join the fiber. Currently, the channel consumer is expected to always join the producer